src/utils/style-formatter.js

```diff
@@ -80,33 +80,39 @@ export default {
     button.innerHTML = '';
     button.appendChild(text);
 
-    // Calculations on font size.
-    // 1. Setting font size to height of button container
-    text.style.fontSize = `${button.clientHeight}px`;
-    // 2. Adjust the font size to the height ratio between button container and text box
-    let newFontsize = (button.clientHeight / text.offsetHeight) * button.clientHeight;
-    text.style.fontSize = `${newFontsize}px`;
-    // 3. Adjust the font size to the width ratio between button container and text box
-    if (text.offsetWidth > button.clientWidth) {
-      newFontsize *= button.clientWidth / text.offsetWidth;
-    }
-    // 4. Setting final font size by scaling with the font size from the layout + other font styling
-    const fontScale = font.size || DEFAULTS.FONT_SIZE;
-    if (font.style && font.style.italic) {
-      if (hasIcon) {
-        text.style.fontSize = `${Math.max(newFontsize * fontScale * 0.84, 8)}px`;
-        text.children[0].style.marginRight = `${text.offsetWidth * 0.04}px`;
-        text.children[1].style.marginRight = `${text.offsetWidth * 0.04}px`;
-      } else {
-        text.style.fontSize = `${Math.max(newFontsize * fontScale * 0.9, 8)}px`;
-        text.children[0].style.marginRight = `${text.offsetWidth * 0.02}px`;
+    // font size adapts to the size of the box and the length of text
+    if (Object.keys(font).length === 0 || font.sizeBehavior === 'responsive') {
```

Comment:
is this check possible? 
```js
Object.keys(font).length === 0
```
According to `src/object-properties.js` you are setting a default value for entire `font` right? which includes default value for `font.sizeBehavior` that makes it to be 'responsive' always (if i'm not mistaken)

Reply:
When `resetAll` from StylingPanel is called all font properties will be deleted and font item will be an empty object `font: {}`. We want to be sure it takes the first option which is the default one(responsive).

Reply:
ok then, this issue goes back to styling panel, that reset should not empty any of the layout property -> it should reset it back to where it was? (in case of default values like this one IMO || at lease it might has a kind rigid default values at the base level to some extent that does not hurts us with checks like this)